resource/filter.go

```diff
@@ -167,3 +177,24 @@ func (f Filter) matches(r *Resource) bool {
 	}
 	return false
 }
+
+func (f *StringFilter) matches(s string) (matched bool, err error) {
+	ok, err := regexp.MatchString(f.Pattern, s)
+	if err == nil && f.Negate {
```

Comment:
I'd prefer simple error checking constructs for readability:
``
```suggestion
	if err != nil {
	  return false, err
	}
	
	if f.Negate {
	  return !ok, nil
	}
	
	return ok, nil
	
```